cubeviz/layout.py

```diff
@@ -74,23 +77,20 @@ def __init__(self, session=None, parent=None):
         self.ui = load_ui('layout.ui', self,
                           directory=os.path.dirname(__file__))
 
-        # Create the views and register to the hub.
-        self.single_view = WidgetWrapper(CubevizImageViewer(self.session, cubeviz_layout=self), tab_widget=self)
-        self.left_view = WidgetWrapper(CubevizImageViewer(self.session, cubeviz_layout=self), tab_widget=self)
-        self.middle_view = WidgetWrapper(CubevizImageViewer(self.session, cubeviz_layout=self), tab_widget=self)
-        self.right_view = WidgetWrapper(CubevizImageViewer(self.session, cubeviz_layout=self), tab_widget=self)
-        self.specviz = WidgetWrapper(SpecVizViewer(self.session), tab_widget=self)
+        self.cube_views = []
 
-        self.single_view._widget.register_to_hub(self.session.hub)
-        self.left_view._widget.register_to_hub(self.session.hub)
-        self.middle_view._widget.register_to_hub(self.session.hub)
-        self.right_view._widget.register_to_hub(self.session.hub)
-        self.specviz._widget.register_to_hub(self.session.hub)
+        # Create the cube viewers and register to the hub.
+        for _ in range(DEFAULT_NUM_SPLIT_VIEWERS + 1):
+            ww = WidgetWrapper(CubevizImageViewer(
+                    self.session, cubeviz_layout=self), tab_widget=self)
+            self.cube_views.append(ww)
+            ww._widget.register_to_hub(self.session.hub)
 
-        self.all_views = [self.single_view, self.left_view, self.middle_view, self.right_view]
+        # Create specviz viewer and register to the hub.
+        self.specviz = WidgetWrapper(SpecVizViewer(self.session), tab_widget=self)
+        self.specviz._widget.register_to_hub(self.session.hub)
 
-        # TODO: determine whether to rename this or get rid of it
-        self.cube_views = self.all_views
+        self.single_view = self.cube_views[0]
         self.split_views = self.cube_views[1:]
 
         self._synced_checkboxes = [
```

Comment:
While this was not generalized, we plan to move the sync checkboxes to the toolbars of the viewers themselves, so it doesn't make sense to fix this here yet.